### ISSUE TITLE
I like the structure

### ISSUE THREAD

**Issue by didlie** [RangerMauve/hyperswarm-proxy#1]
Two issues: 
1) network nodes cannot be fully connected. There must be a network topography.
2) pub sub is not.the goal. Real-time streaming across the distributed network, without server intervention, is the goal... Within in a pub-sub landscape.

**Comment by RangerMauve** [comment]
Yeah, with regards to minimizing connections, you can check out [mostly-minimal-spanning-tree](https://github.com/RangerMauve/mostly-minimal-spanning-tree) which creates a network without partitions where each node has one or two connections on average.

It should be possible to integrate, but I haven't had time to do it yet. PRs are very welcome. You can check out how @geut integrated it with [discovery-swarm-webrtc](https://github.com/geut/discovery-swarm-webrtc/tree/tinchoz49/add-mmst).

Not sure what you mean about the pubsub comment.
